data_entry/views.py

```diff
@@ -1,17 +1,103 @@
 from django.shortcuts import render
 from .models import *
+import datetime
 
+"""
+    Home Page
+
+"""
 def home(request):
     context = {}
     return render(request, 'data_entry/home.html', context)
 
+
+"""
+    Data Log Page
+
+    if a POST request is received:
+        Check if the entry is being loaded -> Get entry_id and display data
+        Check if the entry is being updated -> Update fields and save entry
+    otherwise load empty data log form
+
+"""
 def collectData(request):
     if request.method == 'POST':
+        # if the request contains an entry ID
+        # an entry is being updated or loaded to the form
         if request.POST['entry_id']:
-            entry_id = request.POST['entry_id']
-            entry = Test.objects.get(id=entry_id)
-            context = {'entry': entry}
-            return render(request, 'data_entry/collectdata.html', context)
+            # check if the entry is being updated or loaded
+            method = request.POST['method']
+            # entry is being loaded
+            if method == "load":
+                entry_id = request.POST['entry_id']
+                entry = Test.objects.get(id=entry_id)
+                # format personnel array to a string; needed to display data correctly
+                personnel = entry.personnel
+                personnel = ','.join(personnel)
+
+                # format date; needed to display correct date
+                date_time_str = str(entry.created_at)
+                date_time_str = date_time_str[:-6]
+                date_time_obj = datetime.datetime.strptime(date_time_str, '%Y-%m-%d %H:%M:%S.%f')
+                date = str(date_time_obj.date())
+                context = {
+                    'entry': entry,
+                    'personnel': personnel,
+                    'date': date,
+                    }
+                return render(request, 'data_entry/collectdata.html', context)
+            # entry is being updated
+            else:
+                entry_id = request.POST['entry_id']
+                entry = Test.objects.get(id=entry_id)
+
+                # update fields
+                entry.weather = request.POST['weather']
```

Comment:
@lealal I love this, makes adding new parameters as easy as copying one of these lines and adding the new field

Reply:
Right!?